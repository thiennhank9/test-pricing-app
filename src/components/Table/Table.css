table {
  width: 100%;
}

table,
th,
td {
  border: 1px solid black;
  border-collapse: collapse;
  text-align: center;
  min-width: 100px;
  height: 30px;
}

td.weak {
  background-color: #c5c6d0;
}

td.high-light {
  background-color: #999999;
}

@media screen and (max-width: 768px) {
  table,
  th,
  td {
    min-width: unset;
  }
}
