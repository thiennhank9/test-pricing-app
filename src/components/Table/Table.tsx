import React from "react";
import "./Table.css";

interface TableProps {
  topHeaders: Array<any>;
  leftHeaders: Array<any>;
  data: Array<Array<any>>;
  renderCell(cell: any, rowIndex: number, colIndex: number): JSX.Element;
  onMouseOutTableBody(event: React.MouseEvent): void;
}

function Table(props: TableProps) {
  const {
    topHeaders,
    data,
    renderCell,
    onMouseOutTableBody,
    leftHeaders,
  } = props;

  return (
    <table>
      <thead>
        <tr>
          {topHeaders.map((header) => (
            <th key={header}>{header}</th>
          ))}
        </tr>
      </thead>
      <tbody onMouseOut={onMouseOutTableBody}>
        {data.map((row: Array<any>, rowIndex: number) => (
          <tr key={rowIndex}>
            <th>{leftHeaders[rowIndex]}</th>
            {row.map((cell: any, colIndex: number) =>
              renderCell(cell, rowIndex, colIndex)
            )}
          </tr>
        ))}
      </tbody>
    </table>
  );
}

export default Table;
