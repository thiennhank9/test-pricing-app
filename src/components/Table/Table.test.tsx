import TestRenderer from "react-test-renderer";
import Table from "./Table";

describe("Table", () => {
  test("render table", () => {
    const testRenderer = TestRenderer.create(
      <Table
        topHeaders={["1", "2", "3"]}
        leftHeaders={["1", "2", "3"]}
        data={[
          ["1", "2"],
          ["1", "2"],
          ["1", "2"],
        ]}
        renderCell={(cell, rowIndex, colIndex) => (
          <td key={`${rowIndex}-${colIndex}`}>{cell}</td>
        )}
        onMouseOutTableBody={jest.fn()}
      />
    );
    const testInstance = testRenderer.root;
    expect(testRenderer.toJSON()).toMatchSnapshot();
    const headers = testInstance.findAllByType("th");
    const cells = testInstance.findAllByType("td");
    expect(headers.length).toBe(6);
    expect(cells.length).toBe(6);
  });
});
