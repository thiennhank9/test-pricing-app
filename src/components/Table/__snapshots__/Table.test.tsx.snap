// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table render table 1`] = `
<table>
  <thead>
    <tr>
      <th>
        1
      </th>
      <th>
        2
      </th>
      <th>
        3
      </th>
    </tr>
  </thead>
  <tbody
    onMouseOut={[MockFunction]}
  >
    <tr>
      <th>
        1
      </th>
      <td>
        1
      </td>
      <td>
        2
      </td>
    </tr>
    <tr>
      <th>
        2
      </th>
      <td>
        1
      </td>
      <td>
        2
      </td>
    </tr>
    <tr>
      <th>
        3
      </th>
      <td>
        1
      </td>
      <td>
        2
      </td>
    </tr>
  </tbody>
</table>
`;
